cuda_core/examples/device_properties.py

```diff
@@ -0,0 +1,169 @@
+    print(f"- Max 3D texture height: {properties.maximum_texture3d_height}")
+    print(f"- Max 3D texture depth: {properties.maximum_texture3d_depth}")
+    print(f"- Alternate max 3D texture width: {properties.maximum_texture3d_width_alternate}")
+    print(f"- Alternate max 3D texture height: {properties.maximum_texture3d_height_alternate}")
+    print(f"- Alternate max 3D texture depth: {properties.maximum_texture3d_depth_alternate}")
+    print(f"- Max cubemap texture width or height: {properties.maximum_texturecubemap_width}")
+    print(f"- Max layers in a cubemap layered texture: {properties.maximum_texturecubemap_layered_layers}")
+    print(f"- Max cubemap layered texture width or height: {properties.maximum_texturecubemap_layered_width}")
+    print(f"- Texture base address alignment requirement: {properties.texture_alignment}B")
+    print(f"- Pitch alignment requirement for 2D texture references bound to pitched memory: \
+          {properties.texture_pitch_alignment}B")
+    print(f"- Support for memory pools: {_bool_str(properties.memory_pools_supported)}")
+    print(f"- Bitmask of handle types supported with memory pool-based IPC: \
+          0b{properties.mempool_supported_handle_types:032b}")
+    print(f"- Multi-GPU board: {_bool_str(properties.multi_gpu_board)}")
+    print(f"- Multi-GPU board group ID: {properties.multi_gpu_board_group_id}")
+    print(f"- Support for switch multicast and reduction operations: {_bool_str(properties.multicast_supported)}")
+    print(f"- Number of multiprocessors: {properties.multiprocessor_count}")
+    print(f"- NUMA configuration: {properties.numa_config}")
+    print(f"- NUMA node ID of GPU memory: {properties.numa_id}")
+    print(f"- Support for coherently accessing pageable memory: {_bool_str(properties.pageable_memory_access)}")
+    print(f"- Access pageable memory via host's page tables: \
+          {_bool_str(properties.pageable_memory_access_uses_host_page_tables)}")
+    print(f"- PCI bus ID: {properties.pci_bus_id}")
+    print(f"- PCI device (slot) ID: {properties.pci_device_id}")
+    print(f"- PCI domain ID: {properties.pci_domain_id}")
+    print(f"- Support for registering memory that must be mapped to GPU as read-only: \
+          {_bool_str(properties.read_only_host_register_supported)}")
+    print(f"- Amount of shared memory per block reserved by CUDA driver: \
+          {properties.reserved_shared_memory_per_block}B")
+    print(f"- Support for sparse CUDA arrays and sparse CUDA mipmapped arrays: \
+          {_bool_str(properties.sparse_cuda_array_supported)}")
+    print(f"- Using TCC driver: {_bool_str(properties.tcc_driver)}")
+    print(f"- Constant memory available: {properties.total_constant_memory}B")
+    print(f"- Support for unified address space with host: {_bool_str(properties.unified_addressing)}")
+    print(f"- Support for virtual memory management: {_bool_str(properties.virtual_memory_management_supported)}")
+
+
+ndev = system.num_devices
```

Comment:
Could you please indent everything from here to the end?

```python
def show_device_properties():
    ... your code here ...


if __name__ == "__main__":
    assert len(sys.args) == 1, "no command-line arguments expected"
    show_device_properties()
```

The main difference is that tooling can import this without running anything other than the imports.

Reply:
Yes, will do. Thank you for great comments, Ralf @rwgk!